### ISSUE TITLE
Android: Reopen game with background service after close it in task administartion bar will cause black screen

### ISSUE THREAD

**Issue by tiantian20007** [cocos2d/cocos2d-x#2988]
try my example:  https://github.com/tiantian20007/cocos2dx-jpush
I just add a service to the example project HelloCpp

1 Start a service[http://developer.android.com/reference/android/app/Service.html] in cocos2dx project, any service that will continue to run when you close the app in the task administration bar.
2 Start the game on your android device. 
3 Press home button and get to desktop.
4 Long press on the home button, and close your game.

Then, on most device, the main activity[http://developer.android.com/reference/android/app/Activity.html] of your game will be invoked by onDestroy, but the process of your game will not be terminated because the service you started.
(On some device, the process and service will be terminated too, and wo can do this by the [Force to quit] or [强制退出], in this case, we can't reproduce this issue)

5 You tap your game icon on desktop to restart it: the game will show up with a black screen
And flood by "GL_INVALID_VALUE GL_INVALID_OPERATION cocos2d-x debug info(12583): OpenGL error 0x0502 in [...]/cocos2dx/sprite_nodes/CCSprite.cpp draw 589" in logs.

in conclusion:
I close my game manually, but the service my game started keep the process alive;
When I tap the game icon and try to restart it, the process is not restarted actually. Some openGL state is not right for I am not recreating it and I did not clean up the state or clean resources when our activity.onDestory is invoked.

So the solution maybe that we should add some openGL clean up stuff in activity.onDestory ? Any ideas?

some relative discussion: 
http://www.cocos2d-x.org/boards/6/topics/27903

**Comment by godyZ** [comment]
Please, could you open a bug in redmine in future? We are in process of migrating all the issues to redmine and github issue won't be available after August. The issue is migrated to http://www.cocos2d-x.org/issues/2575
